### ISSUE TITLE
TensorRT Deconvolution in INT8 mode gives obviously wrong result

### ISSUE THREAD

**Issue by estherxue** [NVIDIA/TensorRT#1694]
I used pytorch-quantization to quantize a one-layer network (one Deconvolution layer) and then I passed the model to TensorRT and turned on the explicit quantization mode. I constructed a special weight which is filled with ones except some elements which are set to 127. I also constructed special input. The expected output for TensorRT INT8 model at [0, 0, 0, 0] is 200 and this is what pytorch-quantization model gives but TensorRT model gives 2.57. Could you please explain why TensorRT gives this result? Following is my code and environment in which the code runs.
```
from platform import python_revision
import torch
import torch.nn as nn
import numpy as np


from pytorch_quantization import nn as quant_nn
from pytorch_quantization import calib
from pytorch_quantization.tensor_quant import QuantDescriptor
from pytorch_quantization import quant_modules

quant_modules.initialize()

quant_desc_input = QuantDescriptor(calib_method='max')
quant_nn.QuantConv2d.set_default_quant_desc_input(quant_desc_input)
quant_nn.QuantLinear.set_default_quant_desc_input(quant_desc_input)
quant_nn.QuantConvTranspose2d.set_default_quant_desc_input(quant_desc_input)



class Net(torch.nn.Module):
    def __init__(self):
        super().__init__()
        self.deconv = torch.nn.ConvTranspose2d(192, 100, 4, stride=2, padding=1, bias=False, output_padding=0)
        construct_w = np.ones((200, 100, 4, 4), dtype=np.float32)
        construct_w[:, -1, -1, -1] = 127
        self.deconv.weight = nn.Parameter(torch.tensor(construct_w))

    def forward(self, x):
        out = self.deconv(x)
        return out


model = Net()
model.eval()

# FP forward
dummy_input = torch.tensor(np.tile(np.ones(120).reshape(1, 1, 1, 120), (200, 68, 1)).astype(np.float32))
dummy_input[-1, -1, -1, -1] = 127

def collect_stats(model, data, num_batches):
    """Feed data to the network and collect statistic"""

    # Enable calibrators
    for name, module in model.named_modules():
        if isinstance(module, quant_nn.TensorQuantizer):
            if module._calibrator is not None:
                module.disable_quant()
                module.enable_calib()
            else:
                module.disable()

    model(data)

    # Disable calibrators
    for name, module in model.named_modules():
        if isinstance(module, quant_nn.TensorQuantizer):
            if module._calibrator is not None:
                module.enable_quant()
                module.disable_calib()
            else:
                module.enable()

def compute_amax(model, **kwargs):
    # Load calib result
    for name, module in model.named_modules():
        if isinstance(module, quant_nn.TensorQuantizer):
            if module._calibrator is not None:
                if isinstance(module._calibrator, calib.MaxCalibrator):
                    module.load_calib_amax()
                else:
                    module.load_calib_amax(**kwargs)
            print(F"{name:40}: {module}")
    # model.cuda()


data = dummy_input
with torch.no_grad():
    collect_stats(model, data, num_batches=1)
    compute_amax(model, method="percentile", percentile=100)

out = model(dummy_input)
pytorch_out = out[:]

quant_nn.TensorQuantizer.use_fb_fake_quant = True

torch.onnx.export(
    model, dummy_input, "test.onnx", verbose=True, opset_version=13, enable_onnx_checker=False)

import tensorrt as trt
TRT_LOGGER = trt.Logger(trt.Logger.VERBOSE)
builder = trt.Builder(TRT_LOGGER)
builder.max_batch_size = 1

network_creation_flag = 1 << int(trt.NetworkDefinitionCreationFlag.EXPLICIT_BATCH) | 1 << int(trt.NetworkDefinitionCreationFlag.EXPLICIT_PRECISION)
network = builder.create_network(network_creation_flag)
parser = trt.OnnxParser(network, TRT_LOGGER)
with open('test.onnx', 'rb') as onnx_model:
    parser.parse(onnx_model.read())
for i in range(parser.num_errors):
    error = parser.get_error(i)
    print('error code', error.code)

config = builder.create_builder_config()
config.max_workspace_size = 1 << 30
config.flags |= 1 << int(trt.BuilderFlag.INT8)
engine = builder.build_engine(network, config)

with open('engine_deconv_q.bin', 'wb') as f:
    f.write(engine.serialize())


import pycuda.autoinit
import pycuda.driver as cuda

def infer(engine, img):
    h_input = cuda.pagelocked_empty(trt.volume(engine.get_binding_shape(0)), dtype=np.float32)
    img = img.reshape(-1)
    for i in range(img.shape[0]):
        h_input[i] = img[i]
    h_output = cuda.pagelocked_empty(trt.volume(engine.get_binding_shape(1)), dtype=np.float32)
    # Allocate device memory for inputs and outputs.
    d_input = cuda.mem_alloc(h_input.nbytes)
    d_output = cuda.mem_alloc(h_output.nbytes)
    with engine.create_execution_context() as context:
        # Transfer input data to the GPU.
        cuda.memcpy_htod(d_input, h_input)
        # Run inference.
        context.execute_v2(bindings=[int(d_input), int(d_output)])
        # Return the host output.
        cuda.memcpy_dtoh(h_output, d_output)

    return h_output

trt_out = infer(engine, data.cpu().numpy())
trt_out = trt_out.reshape(pytorch_out.shape)
arr_a = model.deconv.weight.data[:, 0, 1, 1].cpu().numpy()
arr_b = data[0, :, 0, 0].cpu().numpy()
print(np.dot(arr_a, arr_b))
print(trt_out[0, 0, 0, 0])
print(pytorch_out[0, 0, 0, 0])
```
Below is my environment:

TensorRT Version: 8.2.0.6
GPU Type: Tesla T4
Nvidia Driver Version: 470.63.01
CUDA Version: 11.4
CUDNN Version: 8.2.4
Operating System + Version: Ubuntu 18.04
Python Version (if applicable): 3.7.11
PyTorch Version (if applicable): 1.8.0

**Comment by spivakoa** [comment]
Hey, did you fine the source of the problem?

**Comment by ttyio** [comment]
@estherxue  @spivakoa , sorry quantization toolkit misaligned with TRT implementation for `convTranspose`. Could you try quant axis 0 for deconv?

         QUANT_DESC_8BIT_CONVTRANSPOSE2D_WEIGHT_PER_CHANNEL = QuantDescriptor(num_bits=8, axis=(0))

We will fix the code and document, thanks!

**Comment by nvpohanh** [comment]
Closing this for now due to >14 days with no response. Please feel free to reopen if the issue still exists. Thanks!
